### ISSUE TITLE
copyDirRecur says "does not follows symlinks" which is not correct

### ISSUE THREAD

**Issue by nh2** [mrkkrp/path-io#49]
`copyDirRecur` [docs](https://hackage.haskell.org/package/path-io-1.5.0/docs/Path-IO.html#v:copyDirRecur) say

> Copies a directory recursively. **It does not follow symbolic links**

But that its wrong. The logic it implements is "does not descend into symlinks to directories, but copies symlinks to files as if they were files" (and the latter part is usually referred to as "follows symlinks").

## Details and example

The call order is `copyDirRecur` -> `copyDirRecurGen` -> `listDirRecur`.

That calls `isDirSymLink` in
https://github.com/mrkkrp/path-io/pull/23/commits/268fdf680e365e5325f2f4bd7921df7d224c34e0#diff-06c7d82a14aee6aafd149f432cc2dc17R367
`isDirSymLink` checks only if a _directory_ is a symlink. If yes, it won't recurse into it.

It will copy symlinks to files as if they were normal files (so it follows them).

That means that `copyDirRecur [reldir|testdir|] [reldir|testdir2|]` copies
```
testdir
├── subdir
│   └── subdirsubfile
├── subdirlink -> subdir
├── testfile
└── testlink -> testfile
```
into
```
testdir2
├── subdir
│   └── subdirsubfile
├── subdirlink
├── testfile
└── testlink
```

As we can see, `testlink` is now a regular file. `subdirlink` is now a regular directory, but contains no contents.

## What to do instead

It's not clear to me whether the logic is not as intended or the docs are wrong.

But I can't come up with a use case for the current logic, so I suspect the logic is wrong.

For reference, this is what `cp -r` does:

```
% cp -r testdir testdir3
% tree testdir3
testdir3
├── subdir
│   └── subdirsubfile
├── subdirlink -> subdir
├── testfile
└── testlink -> testfile
```

It does not follow symlinks (neither to dirs nor files), but preserves them.

Maybe `copyDirRecur` was intended to behave like this?

**Comment by nh2** [comment]
CC @harendra-kumar from https://github.com/mrkkrp/path-io/pull/23

**Comment by harendra-kumar** [comment]
@nh2 I primarily fixed the traversal behavior of `listDirRecur`. Copy uses `listDirRecur` and therefore followed the same traversal behavior but as you found out the copying behavior of symlinks should also be fixed so that it does not follow symlinks in that case as well and provides a consistent "does not follow symlinks" behavior. We missed that in copy case.

I think this is a bug and should be fixed. The copy behavior for "does not follow symlinks" should be as you suggested above.

**Comment by mrkkrp** [comment]
I'll accept a PR fixing this.

**Comment by mrkkrp** [comment]
@harendra-kumar @nh2 Would any of you be able to find time to put together a PR? I may have time eventually but not very soon.

**Comment by sriduth** [comment]
If no one has started on this yet, I can give it a shot.

**Comment by harendra-kumar** [comment]
@sriduth you can go ahead unless @nh2 already has something on it.

**Comment by nh2** [comment]
@sriduth Please go ahead! :+1:

**Comment by sriduth** [comment]
Hello everyone! I've opened an initial PR - please share your thoughts and feedback.

**Comment by mrkkrp** [comment]
Now `copyDirRecur` and `copyDirRecur'` work exactly like `cp`. @nh2

**Comment by mrkkrp** [comment]
Published 1.6.0 with the fix.
